lib/routes/token.js

```diff
@@ -89,6 +101,60 @@ function confirmRefreshToken(params) {
   });
 }
 
+function confirmJwt(params) {
+  var assertion = params.assertion;
+  logger.debug('jwt.confirm', assertion);
+
+  return JWTOOL.verify(assertion).catch(function(err) {
+    logger.info('jwt.invalid.verify', err.message);
+    throw AppError.invalidAssertion();
+  }).then(function(payload) {
+    logger.verbose('jwt.payload', payload);
+
+    var client = SERVICE_CLIENTS[payload.iss];
+
+    if (!client) {
+      logger.debug('jwt.client.unknown', payload.iss);
+      throw AppError.unknownClient(payload.iss);
+    }
+
+    var uid = payload.sub;
+    if (!uid || !(uid.length === 32 && HEX_STRING.test(uid))) {
+      logger.debug('jwt.invalid.sub', uid);
+      throw AppError.invalidAssertion();
+    }
```

Comment:
In #328 you proposed `"id@accounts.firefox.com"` for the `sub` field, but this just uses the raw uid.  Any comments on which is superior and why?

(FWIW I'm generally in favour of using namespaced identifeirs, but there's no real ambiguity here so I think the plain id is fine as well.  The only thing we'd do if you gave us an unexpected domain in the identifier would be return an error)